rosbag2_py/src/rosbag2_py/_transport.cpp

```diff
@@ -153,12 +153,13 @@ class Player
 class Recorder
 {
 private:
-  std::promise<void> cancel_;
+  rclcpp::executors::SingleThreadedExecutor * exec_;
```

Comment:
please avoid using bare pointers in modern C++ - if we were to use a pointer, I would suggest `unique_ptr`. In this case, though, it should be perfectly safe to use a non-pointer member object.
```suggestion
  rclcpp::executors::SingleThreadedExecutor exec_;
```